feat/injective gateway

vic-en opened the pull request:
**Before submitting this PR, please make sure**:

- [x] Your code builds clean without any errors or warnings
- [x] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request**:
This is the main branch with injective for gateway. We have added the following routes:

```
/injective/balances
/injective/block/current
/injective/poll
/injective/transfer/to/bank
/injective/transfer/to/sub
/clob/markets
/clob/orderBook
/clob/ticker
/clob/orders
/clob/estimateGas
```
Look at test-helpers/curl/curl.sh for any commands with injective in it. That should cover most of the routes. You must first add a wallet for the injective chain. This can be done with an ethereum private key. Then most routes require the injective public address, not the ethereum public address.

Injective has a bank account and subaccounts. The bank account can hold assets, subaccounts can trade assests. If you want to trade you have to move some tokens from the bank account to the sub accounts. We have two routes to move assets between the accounts.

You should be able to get some testnet assets here: https://testnet.injective.exchange/ and view your bank and sub accounts.

**Tests performed by the developer**:
- Basic curl tests.


**Tips for QA testing**:
- Test all curl endpoints and confirm they work as expected.

rapcmia commented:
PR update:
- Testnet funds sent by Injective team and can be requested [here](https://testnet.faucet.injective.network/)
- Setup development client + source gateway(PR19)
  - Connection successful
  - gateway list now displays Injection with no tier (must be `BRONZE`)
     ![image](https://user-images.githubusercontent.com/73840223/219646444-e4e73c4f-1647-451b-a92d-a167aedd4171.png)
- CURL test
  - add_injective_key.json ok
  - injective_balances.json ok
  - injective_poll.json ok 
  - injective_transfer_to_bank.json ok 
  - injective_transfer_to_sub.json ok


Hi @vic-en both `injective_poll.json` and `injective_delete_order.json` have the same message  `BatchUpdateOrders` also did not open a limit order nor delete. Just checking if this is expected? 

Steps to reproduce 
- Steps are also applicable with post order 
     - Create a limit order on the testnet and curl test for delete order
     - Txhash created `6B5D6979278EDEA2F70F5C3343BF0DFB62C8F061B7B6E9BA83F57828B5FE818A`
     - Noticed that the message was about `BatchUpdateOrders`, see link [here](https://testnet.explorer.injective.network/transaction/6B5D6979278EDEA2F70F5C3343BF0DFB62C8F061B7B6E9BA83F57828B5FE818A/)
     - Tried to manually delete the order from the testnet exchange
        - Txhash created `58C8B61B2E8B551DB831FCE97E311E080DCFA1C5462CEB79381FCEE48C1F4FBF`
        - Notice that the message was CancelSpotOrder, see recording below:
        - Should it be the same behavior for the `injective_delete_order.json`?
        
https://user-images.githubusercontent.com/73840223/219647158-f72499a1-5a58-4538-b4b4-ac76ed07f7f1.mp4

vic-en commented:
@rapcmia The transaction hash and orderId are different. 
You poll using transaction hash.
You have to use the orderId to delete the order. OrderId can be seen in response from `curl -s -X GET -k --key $GATEWAY_KEY --cert $GATEWAY_CERT "https://localhost:15888/clob/orders?chain=injective&network=mainnet&connector=injective&market=INJ-USDT&address=XXXX" | jq`, using the your market and address.

aarmoa commented:
@vic-en please solve the merging conflicts and the failing tests

fengtality commented on src/templates/root.yml:
```diff
@@ -87,3 +87,11 @@ configurations:
     configurationPath: pancakeswap.yml
     schemaPath: pangolin-schema.json
+
+  $namespace injectiveCLOB:
```
we are trying to set a standard of 1 template file per connector. Can you move the `gasLimitEstimate` variable from injectiveCLOB into the injective.yml template so that injectiveCLOB can be removed?

fengtality replied:
i know injective is both a chain and connector in the Gateway architecture, but I think users would prefer to have all the Injective configs in one file, and not two.

vic-en replied:
done

fengtality commented on jest.config.js:
```diff
@@ -36,4 +37,7 @@ module.exports = {
   setupFilesAfterEnv: ['<rootDir>/test/setupTests.js'],
   globalSetup: '<rootDir>/test/setup.ts',
   globalTeardown: '<rootDir>/test/teardown.ts',
+  moduleNameMapper: {
```
this change causes many tests to fail:
```
Configuration error:
    
    Could not locate module eccrypto mapped as:
    /Users/feng/Code/gateway/test/mock/eccrypto-mock.js.
    
    Please check your configuration for these entries:
    {
      "moduleNameMapper": {
        "/eccrypto/": "/Users/feng/Code/gateway/test/mock/eccrypto-mock.js"
      },
      "resolver": undefined
    }
```

vic-en replied:
fixed

fengtality commented on src/app.ts:
```diff
@@ -72,28 +65,6 @@ gatewayApp.get('/', (_req: Request, res: Response) => {
   res.status(200).json({ status: 'ok' });
 });
 
-interface ConnectorsResponse {
```
this seems like a cleanup change unrelated to injective?

vic-en replied:
No, it's not.
It's not needed as well. `ConnectorsRoutes.router` does the same and is used by the client.
So we removed it.
However, I can undo it if there's a strict requirement.

fengtality commented:
Also, the Injective `wallet-ts` library introduces many libs with unmet peer dependencies:
```
warning "@injectivelabs/sdk-ts > @improbable-eng/grpc-web-node-http-transport@0.15.0" has unmet peer dependency "@improbable-eng/grpc-web@>=0.13.0".
warning "@injectivelabs/sdk-ts > @improbable-eng/grpc-web-react-native-transport@0.15.0" has unmet peer dependency "@improbable-eng/grpc-web@>=0.13.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets@0.19.12" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @toruslabs/torus-embed@1.22.1" has unmet peer dependency "@babel/runtime@7.x".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-alpha@0.1.8" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-avana@0.1.11" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-backpack@0.1.12" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-bitkeep@0.3.17" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-bitpie@0.5.16" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-blocto@0.5.20" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-brave@0.1.15" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-censo@0.1.2" has unmet peer dependency "@solana/web3.js@^1.61.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-clover@0.4.17" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-coin98@0.5.18" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-coinbase@0.1.16" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-coinhub@0.3.16" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-exodus@0.1.16" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-fractal@0.1.6" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-glow@0.1.16" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-huobi@0.1.13" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-hyperpay@0.1.12" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-keystone@0.1.10" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-krystal@0.1.10" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-ledger@0.9.23" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-magiceden@0.1.11" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-mathwallet@0.9.16" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-neko@0.2.10" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-nightly@0.1.13" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-nufi@0.1.14" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-onto@0.1.5" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-particle@0.1.8" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-phantom@0.9.21" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-safepal@0.5.16" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-saifu@0.1.13" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-salmon@0.1.12" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-sky@0.1.13" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-slope@0.5.19" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-solflare@0.6.23" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-sollet@0.11.15" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-solong@0.9.16" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-spot@0.1.13" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-strike@0.1.11" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-tokenary@0.1.10" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-tokenpocket@0.4.17" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-torus@0.11.26" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-trust@0.1.11" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-unsafe-burner@0.1.5" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-walletconnect@0.1.11" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-xdefi@0.1.5" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @toruslabs/torus-embed > @toruslabs/fetch-node-details@5.1.0" has unmet peer dependency "@babel/runtime@7.x".
warning "@injectivelabs/wallet-ts > @toruslabs/torus-embed > @toruslabs/http-helpers@2.2.0" has unmet peer dependency "@babel/runtime@7.x".
warning "@injectivelabs/wallet-ts > @toruslabs/torus-embed > @toruslabs/openlogin-jrpc@1.7.3" has unmet peer dependency "@babel/runtime@7.x".
warning "@injectivelabs/wallet-ts > @toruslabs/torus-embed > @toruslabs/torus.js@5.1.0" has unmet peer dependency "@babel/runtime@7.x".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-alpha > @solana/wallet-adapter-base@0.9.21" has unmet peer dependency "@solana/web3.js@^1.58.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-blocto > @blocto/sdk@0.2.22" has unmet peer dependency "@solana/web3.js@^1.30.2".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-particle > @particle-network/solana-wallet@0.5.6" has unmet peer dependency "@solana/web3.js@^1.50.1".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-particle > @particle-network/solana-wallet@0.5.6" has unmet peer dependency "bs58@^4.0.1".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-salmon > salmon-adapter-sdk@1.1.1" has unmet peer dependency "@solana/web3.js@^1.44.3".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-solflare > @solflare-wallet/sdk@1.2.0" has unmet peer dependency "@solana/web3.js@^1.61.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-sollet > @project-serum/sol-wallet-adapter@0.2.6" has unmet peer dependency "@solana/web3.js@^1.5.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-torus > @toruslabs/solana-embed@0.3.3" has unmet peer dependency "@babel/runtime@7.x".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-walletconnect > @jnwng/walletconnect-solana@0.1.4" has unmet peer dependency "@solana/web3.js@^1.52.0".
warning "@injectivelabs/wallet-ts > @toruslabs/torus-embed > @toruslabs/openlogin-jrpc > @toruslabs/openlogin-utils@1.7.0" has unmet peer dependency "@babel/runtime@7.x".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-fractal > @fractalwagmi/solana-wallet-adapter > @fractalwagmi/popup-connection@1.0.21" has unmet peer dependency "react@^17.0.2 || ^18".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-fractal > @fractalwagmi/solana-wallet-adapter > @fractalwagmi/popup-connection@1.0.21" has unmet peer dependency "react-dom@^17.0.2 || ^18".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-solflare > @solflare-wallet/sdk > @project-serum/sol-wallet-adapter@0.2.0" has unmet peer dependency "@solana/web3.js@^1.5.0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-torus > @toruslabs/solana-embed > @toruslabs/base-controllers@2.7.0" has unmet peer dependency "@babel/runtime@7.x".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-torus > @toruslabs/solana-embed > @toruslabs/http-helpers@3.2.0" has unmet peer dependency "@babel/runtime@^7.x".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-torus > @toruslabs/solana-embed > @toruslabs/openlogin-jrpc@2.13.0" has unmet peer dependency "@babel/runtime@7.x".
warning "@injectivelabs/wallet-ts > @walletconnect/web3-provider > web3-provider-engine > eth-block-tracker > @babel/plugin-transform-runtime@7.21.0" has unmet peer dependency "@babel/core@^7.0.0-0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-torus > @toruslabs/solana-embed > @toruslabs/base-controllers > @toruslabs/openlogin-jrpc@3.0.0" has unmet peer dependency "@babel/runtime@7.x".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-torus > @toruslabs/solana-embed > @toruslabs/openlogin-jrpc > @toruslabs/openlogin-utils@2.13.0" has unmet peer dependency "@babel/runtime@7.x".
warning "@injectivelabs/wallet-ts > @walletconnect/web3-provider > web3-provider-engine > eth-block-tracker > @babel/plugin-transform-runtime > babel-plugin-polyfill-corejs2@0.3.3" has unmet peer dependency "@babel/core@^7.0.0-0".
warning "@injectivelabs/wallet-ts > @walletconnect/web3-provider > web3-provider-engine > eth-block-tracker > @babel/plugin-transform-runtime > babel-plugin-polyfill-corejs3@0.6.0" has unmet peer dependency "@babel/core@^7.0.0-0".
warning "@injectivelabs/wallet-ts > @walletconnect/web3-provider > web3-provider-engine > eth-block-tracker > @babel/plugin-transform-runtime > babel-plugin-polyfill-regenerator@0.4.1" has unmet peer dependency "@babel/core@^7.0.0-0".
warning "@injectivelabs/wallet-ts > @solana/wallet-adapter-wallets > @solana/wallet-adapter-torus > @toruslabs/solana-embed > @toruslabs/base-controllers > @toruslabs/openlogin-jrpc > @toruslabs/openlogin-utils@3.0.0" has unmet peer dependency "@babel/runtime@7.x".
warning "@injectivelabs/wallet-ts > @walletconnect/web3-provider > web3-provider-engine > eth-block-tracker > @babel/plugin-transform-runtime > babel-plugin-polyfill-corejs2 > @babel/helper-define-polyfill-provider@0.3.3" has unmet peer dependency "@babel/core@^7.4.0-0".

```

vic-en commented:
@fengtality I addressed all comments. Pls take a look.
I believe the unit tests should  pass now. 
Just like to add that the unit tests in general need to be debugged for memory leaks. I had to increase the `--max_old_space_size` before the. test suite could complete locally. So I suggest an investigation ticket be created for that.

vic-en commented:
@rapcmia I removed review request by mistake. Pls do necessary reviews. Thanks.

rapcmia commented:
> @rapcmia The transaction hash and orderId are different.
> You poll using transaction hash.
> You have to use the orderId to delete the order. OrderId can be seen in response from `curl -s -X GET -k --key $GATEWAY_KEY --cert $GATEWAY_CERT "https://localhost:15888/clob/orders?chain=injective&network=mainnet&connector=injective&market=INJ-USDT&address=XXXX" | jq`, using the your market and address.
Thanks @vic-en however I think there is something going on with the testnet server and for some reason we cant transfer testnet funds to trading account. Tried new and old wallet still the same even the our curl tests response no txhash atm. Will try later today again

![image](https://user-images.githubusercontent.com/73840223/220524914-1278f1e1-631d-48db-a5eb-db9f4e9cea60.png)

rapcmia reviewed (approved):
LGTM

- Ran curl endpoint tests
  - Port ok 
  - add wallet ok 
  - check balances ok 
  - check poll ok 
  - post order: ok 
 	  - use the script provided in order to get the orderHash used for the orderId on the delete endpoint
  	  - curl -s -X GET -k --key $GATEWAY_KEY --cert $GATEWAY_CERT 'https://localhost:$PORT/clob/orders?chain=injective&network=mainnet&connector=injective&market=INJ-USDT&orderId=0xa508cb32923323679f29a032c70342c147c17d0145625922b0ef22e955c844c0&address=$INJECTIVE_SUBACCOUNTID' | jq
  - delete order ok 
  - transfer to bank ok
  - transfer to sub ok  
- Check client as well and injector_protocol now showing on `gateway list` when running on development source client

nikspz reviewed (dismissed):
LGTM

- Ran curl endpoint tests
  - Port ok 
  - add wallet ok 
  - check balances ok 
  - check poll ok 
  - post order: ok 
 	  - use the script provided in order to get the orderHash used for the orderId on the delete endpoint
  	  - curl -s -X GET -k --key $GATEWAY_KEY --cert $GATEWAY_CERT 'https://localhost:$PORT/clob/orders?chain=injective&network=mainnet&connector=injective&market=INJ-USDT&orderId=0xa508cb32923323679f29a032c70342c147c17d0145625922b0ef22e955c844c0&address=$INJECTIVE_SUBACCOUNTID' | jq
  - delete order ok 
  - transfer to bank ok
  - transfer to sub ok  
- Check client as well and injector_protocol now showing on `gateway list` when running on development source client

nikspz commented:
btw, not sure if that warning message is expected since it's `not showed on development`

gw PR19:

Steps:
1. clone and install development branch
2. clone injective gateway PR19
3. yarn
4. yarn build
5. ./gateway-setup.sh
6. yarn start --passphrase=password
7. pay attention to the docker gateway logs


Actual:
secp256k1 unavailable, reverting to browser version 
![image](https://user-images.githubusercontent.com/83953535/222383284-b21b8722-5f7e-440d-b3c4-29deb67e26a9.png)



current Development gateway:
![image](https://user-images.githubusercontent.com/83953535/222383171-e697b4aa-28a6-4f0a-8678-968fc56196ae.png)

aarmoa commented:
@nikspz this PR does not make any change related to secp256k1. Did you run `yarn install` before the command showed in the screenshot?

nikspz commented:
@aarmoa I used
```
yarn
yarn build
./gateway-setup.sh
yarn start --passphrase=password
```
